packages/dataviews/src/style.scss

```diff
@@ -216,6 +216,10 @@
 	display: block;
 	width: 100%;
 
+	&:has(a) {
```

Comment:
I think the misalignment with the padding won't be great in general.. Maybe we could remove the `overflow: hidden;` in the main container above? 🤔  --cc @jameskoster

Reply:
I agree this could lead to inconsistencies in a single data view which isn't ideal. It also breaks the 4px spacing system. 

As for whether the styles are built-in, I see that as an implementation detail best left to y'all engineers to decide :) I can see that built-in styles could be unhelpful in some situations. As long as it's possible to apply truncation to page / template / pattern titles in certain layouts as required I'm happy.

Either way, the issue this PR aims to fix will continue to exist. @t-hamano perhaps an alternative approach could be to add 4px horizontal padding to `.dataviews-view-grid__primary-field` and reduce the `gap` on `dataviews-view-grid__title-actions` from 8 to 4px?

Reply:
I've got a PR that touches on the same thing but doesn't need to add the padding  https://github.com/WordPress/gutenberg/pull/58814

Reply:
@oandregal I had overlooked that PR 😅 #58814 seems to have a broader discussion going on, so I'm happy to close this PR to prevent cluttering the discussion.